build.rs

```diff
@@ -1,66 +1,64 @@
+use std::env;
 use std::path::Path;
 use std::path::PathBuf;
 use std::process::Command;
-use std::{env, fs, io};
 
 #[cfg(feature = "bundled")]
-fn init_submodule(path: &Path) {
+fn fetch_hwloc(path: &Path, version: &str) {
     if !path.join("Makefile.am").exists() {
         let r = Command::new("git")
-            .args(&["clone", "https://github.com/open-mpi/hwloc"])
+            .args(&[
+                "clone",
+                "https://github.com/open-mpi/hwloc",
+                "--depth",
+                "1",
+                "--branch",
+                version,
+            ])
             .current_dir(path.clone())
             .output();
         match r {
-            Ok(m) => {}
-            Err(e) => println!("Error {}", e),
+            Ok(m) => {
+                if !m.status.success() {
+                    eprintln!("git clone for hwloc returned {}", m.status);
+                    eprintln!("{:?}", m);
+                }
+            }
+            Err(e) => eprintln!("Error cloning hwloc {}", e),
         }
     }
 }
 
 #[cfg(feature = "bundled")]
-fn get_os_from_triple(triple: &str) -> Option<&str> {
-    triple.splitn(3, "-").nth(2)
-}
-
-#[cfg(feature = "bundled")]
-fn compile_hwloc2(build_path: &Path, target_os: &str) -> PathBuf {
+fn compile_hwloc2() -> PathBuf {
     let mut config = autotools::Config::new("hwloc");
     config.reconf("-ivf").make_target("install").build()
 }
 
 fn main() {
-    let required_version = if cfg!(feature = "hwloc-2_8_0") {
-        "2.8.0"
+    let (required_version, source_version) = if cfg!(feature = "hwloc-2_8_0") {
+        ("2.8.0", "v2.8")
     } else if cfg!(feature = "hwloc-2_5_0") {
-        "2.5.0"
+        ("2.5.0", "v2.5")
     } else if cfg!(feature = "hwloc-2_4_0") {
-        "2.4.0"
+        ("2.4.0", "v2.4")
     } else if cfg!(feature = "hwloc-2_3_0") {
-        "2.3.0"
+        ("2.3.0", "v2.3")
     } else if cfg!(feature = "hwloc-2_2_0") {
-        "2.2.0"
+        ("2.2.0", "v2.2")
     } else if cfg!(feature = "hwloc-2_1_0") {
-        "2.1.0"
+        ("2.1.0", "v2.1")
     } else if cfg!(feature = "hwloc-2_0_4") {
-        "2.0.4"
+        ("2.0.4", "v2.x")
     } else {
-        "2.0.0"
+        ("2.0.0", "v2.0")
```

Comment:
hwloc minor/bugfix releases are backwards compatible, and they may fix some bugs so it is better to use the latest one you can afford.

Therefore, I think you can get away with this:

```rust
let source_version = match required_version.split('.')
    .next()
    .expect("No major version in required_version")
{
    "2" => "v2.x",
    other => panic!("Please add support for bundling hwloc v{other}.x"),
};
```